e2e/testcases/helm_sync_test.go

```diff
@@ -62,10 +65,66 @@ func TestPublicHelm(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.Unstructured)
 
 	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+	rs.Spec = v1beta1.RootSyncSpec{
+		SourceFormat: string(filesystem.SourceFormatUnstructured),
+		SourceType:   string(v1beta1.HelmSource),
+		Helm: &v1beta1.HelmRootSync{
+			Namespace: "wordpress",
+			HelmBase: v1beta1.HelmBase{
+				Repo:        "https://charts.bitnami.com/bitnami",
+				Chart:       "wordpress",
+				Version:     "15.2.35",
+				ReleaseName: "my-wordpress",
+				Auth:        configsync.AuthNone,
+				Values: &apiextensionsv1.JSON{
+					Raw: []byte(`{
+  "image": {
+    "digest": "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e",
+    "pullPolicy": "Always"
+  },
+  "wordpressUsername": "test-user",
+  "wordpressEmail": "test-user@example.com",
+  "extraEnvVars": [
+    {"name": "TEST_1", "value": "val1"},
+    {"name": "TEST_2", "value": "val2"}
+  ],
+  "resources": {
+    "requests": {
+      "cpu": "150m",
+      "memory": "250Mi"
+    },
+    "limits": {
+      "cpu": "1",
+      "memory": "300Mi"
+    }
+  },
+  "mariadb": {
+    "primary": {
+      "persistence": {
+        "enabled": false
+      }
+    }
+  },
+  "service": {
+    "type": "ClusterIP"
+  }
+}`),
+				},
+			},
+		},
+	}
+	if nt.IsGKEAutopilot {
+		nt.T.Log("Increasing memory request/limit for helm-sync on Autopilot")
+		rs.Spec.SafeOverride().Resources = []v1beta1.ContainerResourcesSpec{
+			{ // This chart sometimes causes OOMKill on Autopilot with default limit
```

Comment:
While the WordPress Helm chart is not resource-intensive, if OOMKill errors occur in our end-to-end tests, we should probably consider bumping the default limit.

Reply:
I agree, although it typically involves more convincing to increase the default limits

Reply:
Do you know how frequent we are hitting OOMKills and which Autopilot channel it is?

Reply:
It's currently the greatest source of flakiness on autopilot, which is why I decided to take a closer look at this test. See:
- https://testgrid.k8s.io/googleoss-kpt-config-sync-main#autopilot-regular
- https://testgrid.k8s.io/googleoss-kpt-config-sync-main#autopilot-rapid
- https://testgrid.k8s.io/googleoss-kpt-config-sync-main#autopilot-rapid-latest

It appears to be failing roughly 20% of the time or so

Reply:
I see. Let's go with increasing the limit in e2e tests only. We can increase the default when more test cases are failing.